### PR TITLE
EZP-31030: Introduced strict types for SectionService

### PR DESCRIPTION
**Pull request by adamwojs**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31030](https://jira.ez.no/browse/EZP-31030)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `master`
| **BC breaks**      | yes
| **Tests pass**     | yes
| **Doc needed**     | no

This PR introduces strict types for the `\eZ\Publish\API\Repository\SectionService`.

**TODO**:
- [X] Implement feature / fix a bug.
- [X] Implement tests.
- [X] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/API/Repository/SectionService.php [hunk 1]
```diff
@@ -54,14 +54,14 @@ public function updateSection(Section $section, SectionUpdateStruct $sectionUpda
      *
      * @return \eZ\Publish\API\Repository\Values\Content\Section
      */
-    public function loadSection($sectionId);
+    public function loadSection(int $sectionId): Section;
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/SectionService.php]: `@param int $sectionId`

File: eZ/Publish/API/Repository/SectionService.php [hunk 2]
```diff
@@ -108,7 +108,7 @@ public function isSectionUsed(Section $section);
      * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo $contentInfo
      * @param \eZ\Publish\API\Repository\Values\Content\Section $section
      */
-    public function assignSection(ContentInfo $contentInfo, Section $section);
+    public function assignSection(ContentInfo $contentInfo, Section $section): void;
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/SectionService.php]: If you could, please, correct in the meantime `Assigns the content to the given section this method overrides the current assigned section.` to be on one line

**Comment by adamwojs** [comment]
PR updated according @mikadamczyk  and @ViniTou to code review suggestions.

**Comment by adamwojs** [comment]
> Error when assigning content to section and removing section.
> 
> Argument 1 passed to eZ\Publish\SPI\Repository\Decorator\SectionServiceDecorator::loadSection() must be of the type int, string given, called in /home/vagrant/master/vendor/ezsystems/ezplatform-admin-ui/src/lib/Form/DataTransformer/SectionTransformer.php on line 73

@katarzynazawada Fixed in https://github.com/ezsystems/ezplatform-admin-ui/pull/1140

**Pull request merged**